pkg/ingester/ingester.go

```diff
@@ -761,8 +828,31 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 	i.appendedSamplesStats.Inc(int64(stats.succeededSamplesCount))
 	i.appendedExemplarsStats.Inc(int64(stats.succeededExemplarsCount))
 
+	i.metrics.ephemeralIngestedSamples.WithLabelValues(userID).Add(float64(ephemeralStats.succeededSamplesCount))
+	i.metrics.ephemeralIngestedSamplesFail.WithLabelValues(userID).Add(float64(ephemeralStats.failedSamplesCount))
+	i.appendedSamplesStats.Inc(int64(ephemeralStats.succeededSamplesCount))
+
 	group := i.activeGroups.UpdateActiveGroupTimestamp(userID, validation.GroupLabel(i.limits, userID, req.Timeseries), startAppend)
 
+	i.updateMetricsFromPushStats(userID, group, &stats)
+	i.updateMetricsFromPushStats(userID, group, &ephemeralStats)
+
+	db.updatedRatesFromStats(stats.succeededSamplesCount, req.Source)
+	db.updatedRatesFromStats(ephemeralStats.succeededSamplesCount, req.Source)
```

Comment:
Why do we need this new function and can't keep the metrics update in `updateMetricsFromPushStats()`, passing the `req.Source` there?

Reply:
We also need to pass `db` there. We can do that too.